extensions/wikia/Discussions/utils/DiscussionsVarToggler.class.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+class DiscussionsVarToggler {
+
+	// Discussions related wg variables
+	const ENABLE_DISCUSSIONS_NAV = 'wgEnableDiscussionsNavigation';
+	const ENABLE_DISCUSSIONS = 'wgEnableDiscussions';
+	const ENABLE_FORUMS = 'wgEnableForumExt';
+	const ARCHIVE_WIKI_FORUMS = 'wgArchiveWikiForums';
+
+	private $discussionsVarMap;
+	private $cityId;
+
+	public function __construct( $cityId = null ) {
+		$this->cityId = $cityId ?? F::app()->wg->CityId;
+		$this->discussionsVarMap = [
+			self::ENABLE_DISCUSSIONS => null,
+			self::ENABLE_DISCUSSIONS_NAV => null,
+			self::ENABLE_FORUMS => null,
+			self::ARCHIVE_WIKI_FORUMS => null
+		];
+	}
+
+	public function setEnableDiscussions( bool $val ) : DiscussionsVarToggler {
+		$this->discussionsVarMap[self::ENABLE_DISCUSSIONS] = $val;
+		return $this;
+	}
+
+	public function setEnableDiscussionsNav( bool $val ) : DiscussionsVarToggler {
+		$this->discussionsVarMap[self::ENABLE_DISCUSSIONS_NAV] = $val;
+		return $this;
+	}
+
+	public function setEnableForums( bool $val ) : DiscussionsVarToggler {
+		$this->discussionsVarMap[self::ENABLE_FORUMS] = $val;
+		return $this;
+	}
+
+	public function setArchiveWikiForums( bool $val ) : DiscussionsVarToggler {
+		$this->discussionsVarMap[self::ARCHIVE_WIKI_FORUMS] = $val;
+		return $this;
+	}
```

Comment:
That's true, but I think it's pretty limited in scope of what it does. I was thinking of this as more just using a fluent interface, rather than implementing builder specifically. The wikipedia page on the [fluent interface](https://en.wikipedia.org/wiki/Fluent_interface) has a couple of examples that look very much like this (class with fields that can be saved to db).

If it's not a blocker for you, I think I'd prefer to leave this as is for now. I think the API works well for what it's used for, and I'm worried that pulling out the fields into a separate builder class would make using it a little bit more clunky.